rpcs3/Emu/RSX/Common/BufferUtils.cpp

```diff
@@ -755,39 +733,277 @@ namespace
 
 	struct primitive_restart_impl
 	{
+		AVX2_FUNC
+		static
+		std::tuple<u16, u16, u32> upload_u16_swapped_avx2(const void *src, void *dst, u32 count, u16 restart_index)
+		{
+			const __m256i shuffle_mask = _mm256_set_epi8(
+				0xE, 0xF, 0xC, 0xD,
+				0xA, 0xB, 0x8, 0x9,
+				0x6, 0x7, 0x4, 0x5,
+				0x2, 0x3, 0x0, 0x1,
+				0xE, 0xF, 0xC, 0xD,
+				0xA, 0xB, 0x8, 0x9,
+				0x6, 0x7, 0x4, 0x5,
+				0x2, 0x3, 0x0, 0x1);
+
+			auto src_stream = (const __m256i*)src;
+			auto dst_stream = (__m256i*)dst;
+
+			__m256i restart = _mm256_set1_epi16(restart_index);
+			__m256i min = _mm256_set1_epi16(0xffff);
+			__m256i max = _mm256_set1_epi16(0);
+
+			const auto iterations = count / 16;
+			for (unsigned n = 0; n < iterations; ++n)
+			{
+				const __m256i raw = _mm256_loadu_si256(src_stream++);
+				const __m256i value = _mm256_shuffle_epi8(raw, shuffle_mask);
+				const __m256i mask = _mm256_cmpeq_epi16(restart, value);
+				const __m256i value_with_min_restart = _mm256_andnot_si256(mask, value);
+				const __m256i value_with_max_restart = _mm256_or_si256(mask, value);
+				max = _mm256_max_epu16(max, value_with_min_restart);
+				min = _mm256_min_epu16(min, value_with_max_restart);
+				_mm256_storeu_si256(dst_stream++, value);
```

Comment:
This is still incorrect, check the reference pseudocode I gave you. You're supposed to write the index data with the restart index pushed to uint_max, not the value the game was using. In this case, you should write value_with_max_restart.

Reply:
Oops sorry, this is now fixed.